runtime/src/syscalls_impl_arm.rs

```diff
@@ -0,0 +1,92 @@
+use libtock_platform::RawSyscalls;
+
+unsafe impl RawSyscalls for crate::TockSyscalls {
+    unsafe fn yield1([r0]: [*mut (); 1]) {
+        // Safety: This matches the invariants required by the documentation on
+        // RawSyscalls::yield1
+        unsafe {
+            asm!("svc 0",
+                 inlateout("r0") r0 => _, // a1
+                 lateout("r1") _,         // a2
+                 lateout("r2") _,         // a3
+                 lateout("r3") _,         // a4
```

Comment:
The comments still have the risc-v register names

Reply:
Actually, these are the ARM "synonym names" for the registers. In both the ARM and RISC-V `RawSyscalls` implementations, I put the raw register names in the code and alternate names in the comments.

I totally see why this is confusing however. Should I remove the synonyms to avoid the confusion?

Reply:
oh wow, I see. I didn't realize there were synonym names for argument registers! In that case I think this is fine